Metrics.go

```diff
@@ -19,10 +19,10 @@ type (
 		Histograms    map[string]prometheus.Histogram
 		Gauges        map[string]prometheus.Gauge
 		Logger        *logger.Logger
-		countMutex    *sync.Mutex
-		countVecMutex *sync.Mutex
-		histMutex     *sync.Mutex
-		gaugeMutex    *sync.Mutex
+		countMutex    *sync.RWMutex
```

Comment:
Just out of curiosity: was there any reason we had to use a Mutex, and couldn't utilize some higher level mechanism (e.g. channels)?